portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,22 +20,29 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import java.util.ArrayList;
+import java.util.HashMap;
+// import java.util.ArrayList;
 import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  ArrayList<String> messages = new ArrayList<>();
+  HashMap<String, String> messages = new HashMap<String, String>();
+  // ArrayList<String> messages = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    messages.add("This is brought to you by JSON");
-    messages.add("JSON is JavaScript Object Notation");
-    messages.add("It is great for storing data like this");
-
     response.setContentType("application/json;");
     String json = new Gson().toJson(messages);
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String username = request.getParameter("username");
+    String message = request.getParameter("comment-or-question");
+    messages.put(username, message);
```

Comment:
no need to take action now, but what if a client sends in a null username? or a null comment?

Reply:
I could disable null usernames/comments later otherwise likely no exceptions will be thrown if I do not disable null usernames/comments.